cmd/server/app/watch.go

```diff
@@ -0,0 +1,182 @@
+limitations under the License.
+*/
+
+package app
+
+import (
+	"context"
+	"crypto"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/google/trillian"
+	tclient "github.com/google/trillian/client"
+	tcrypto "github.com/google/trillian/crypto"
+	"github.com/google/trillian/merkle/rfc6962/hasher"
+	"github.com/google/trillian/types"
+
+	"github.com/sigstore/rekor/cmd/cli/app"
+	"github.com/sigstore/rekor/pkg/generated/client"
+	"github.com/sigstore/rekor/pkg/log"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"gocloud.dev/blob"
+	_ "gocloud.dev/blob/gcsblob"
+)
+
+// watchCmd represents the serve command
+var watchCmd = &cobra.Command{
+	Use:   "watch",
+	Short: "start a process to watch and record STH's from Rekor",
+	Long:  `start a process to watch and record STH's from Rekor`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		// Setup the logger to dev/prod
+		log.ConfigureLogger(viper.GetString("log_type"))
+
+		// workaround for https://github.com/sigstore/rekor/issues/68
+		// from https://github.com/golang/glog/commit/fca8c8854093a154ff1eb580aae10276ad6b1b5f
+		_ = flag.CommandLine.Parse([]string{})
+
+		host := viper.GetString("rekor_server.address")
+		port := viper.GetUint("rekor_server.port")
+		url := fmt.Sprintf("http://%s:%d", host, port)
+		c, err := app.GetRekorClient(url)
+		if err != nil {
+			return err
+		}
+
+		keyResp, err := c.Tlog.GetPublicKey(nil)
+		if err != nil {
+			return err
+		}
+		publicKey := keyResp.Payload
+		block, _ := pem.Decode([]byte(publicKey))
+		if block == nil {
+			return errors.New("failed to decode public key of server")
+		}
+
+		pub, err := x509.ParsePKIXPublicKey(block.Bytes)
+		if err != nil {
+			return err
+		}
+
+		verifier := tclient.NewLogVerifier(hasher.DefaultHasher, pub, crypto.SHA256)
+		ctx := context.Background()
+		bucketURL := os.Getenv("REKOR_STH_BUCKET")
+		bucket, err := blob.OpenBucket(ctx, bucketURL)
+		if err != nil {
+			return err
+		}
+		defer bucket.Close()
+		tick := time.NewTicker(1 * time.Minute)
+		var last *SignedAndUnsignedLogRoot
+		for {
+			<-tick.C
+			log.Logger.Info("performing check")
+			lr, err := doCheck(c, verifier)
+			if err != nil {
+				log.Logger.Warnf("error verifiying tree: %s", err)
+				continue
+			}
+			log.Logger.Infof("Found and verified state at %d %d", lr.VerifiedLogRoot.TreeSize, lr.VerifiedLogRoot.TimestampNanos)
+			if last != nil && last.VerifiedLogRoot.TreeSize == lr.VerifiedLogRoot.TreeSize {
+				log.Logger.Infof("Last tree size is the same as the current one: %d %d",
+					last.VerifiedLogRoot.TreeSize, lr.VerifiedLogRoot.TreeSize)
+				// If it's the same, it shouldn't have changed but we'll still upload anyway
+				// in case that failed.
+			}
+
+			if err := uploadToGcs(ctx, bucket, lr); err != nil {
+				log.Logger.Warnf("error uploading result: %s", err)
+				continue
+			}
+			last = lr
+		}
+	},
+}
+
+func init() {
```

Comment:
There's a duplicate `init` at the bottom of this file.

Reply:
Thanks!